extensions/resource-deployment/src/ui/modelViewUtils.ts

```diff
@@ -256,24 +295,41 @@ function processFields(fieldInfoArray: FieldInfo[], components: azdata.Component
 	}
 }
 
-export function createFlexContainer(view: azdata.ModelView, items: azdata.Component[], rowLayout: boolean = true): azdata.FlexContainer {
+export function createFlexContainer(view: azdata.ModelView, items: azdata.Component[], rowLayout: boolean = true, width?: string | number, height?: string | number, alignItems?: azdata.AlignItemsType, cssStyles?: CSSStyles): azdata.FlexContainer {
 	const flexFlow = rowLayout ? 'row' : 'column';
-	const alignItems = rowLayout ? 'center' : undefined;
-	const itemsStyle = rowLayout ? { CSSStyles: { 'margin-right': '5px' } } : {};
-	return view.modelBuilder.flexContainer().withItems(items, itemsStyle).withLayout({ flexFlow: flexFlow, alignItems: alignItems }).component();
+	alignItems = alignItems || (rowLayout ? 'center' : undefined);
+	const itemsStyle = rowLayout ? { CSSStyles: { 'margin-right': '5px', } } : {};
+	const flexLayout: azdata.FlexLayout = { flexFlow: flexFlow };
+	if (height) {
+		flexLayout.height = height;
+	}
+	if (width) {
+		flexLayout.width = width;
+	}
+	if (alignItems) {
+		flexLayout.alignItems = alignItems;
+	}
+	let flexComponent = view.modelBuilder.flexContainer().withItems(items, itemsStyle).withLayout(flexLayout).component();
+	if ( cssStyles ) {
+		flexComponent.CSSStyles = Object.assign(flexComponent.CSSStyles || {}, cssStyles);
```

Comment:
It'd be better just to do this before creating the component and assigning it during creation

Reply:
will do.

Reply:
fixed